infrastructre/template.yaml

```
@@ -44,3 +44,11 @@ Resources:
           Value: !Ref 'Environment'
         - Key: Source
           Value: "https://github.com/alphagov/di-account-management-backend/blob/main/infrastructre/template.yaml"
+
+  ######################################
+  # TxMA Event Queue (Dummy for now)
+  ######################################
+  TxMAInputDummyQueue:
+    Type: AWS::SQS::Queue
+    DeletionPolicy: Retain
```

cfn-lint wants these to be explicit. Not sure if Retain is right, but figured I'd go with that for now?

---

I think `Retain` will be annoying when we're testing this in the dev account since we'll end up with lots of queues and Dynamo tables.  We do want it for all the other environments though. I think this is a perfect use case for a `Condition`:

At the top of the file under `Parameters` add in

```
Conditions:
  IsDevelopmentEnvironment: !Equals 
    - !Ref Environment
    - dev
```

Then you can change this (and all the other instances of `Retain`) to 

```suggestion
    DeletionPolicy: !If [IsDevelopmentEnvironment, Delete, Retain]
```

---

Nice, I like this! I'll add that!

---

What do you think about `UpdateReplacePolicy`? Is that just superfluous?

---

Cloudformation seems really unhappy with this 🤔 It's saying it's expecting a string.
I've tried:
```
!Join [ '', !If [IsDevelopmentEnvironment, "Delete", "Retain"] ]
```

But it doesn't seem to like that either, I've seen examples of folks doing this on various github issues, so I'm gonna guess this is cfn-lint freaking out.

I reckon we try and apply this and see what SAM says and adjust the linter from there?